engine/src/main/java/org/pentaho/di/trans/step/BaseStepHelper.java

```diff
@@ -0,0 +1,67 @@
+/*! ******************************************************************************
+ *
+ * Pentaho
+ *
+ * Copyright (C) 2024 by Hitachi Vantara, LLC : http://www.pentaho.com
+ *
+ * Use of this software is governed by the Business Source License included
+ * in the LICENSE.TXT file.
+ *
+ * Change Date: 2029-07-20
+ ******************************************************************************/
+
+
+package org.pentaho.di.trans.step;
+
+import org.json.simple.JSONObject;
+import org.pentaho.di.core.logging.KettleLogStore;
+import org.pentaho.di.core.logging.LogChannelInterface;
+import org.pentaho.di.trans.TransMeta;
+
+import java.util.Map;
+
+public abstract class BaseStepHelper implements StepHelperInterface {
+
+  protected final LogChannelInterface log;
+
+  protected BaseStepHelper() {
+    this.log = KettleLogStore.getLogChannelInterfaceFactory().create( this );
+  }
+
+  /**
+   * Template method for executing an action method. Handles response status and error logging.
+   * Subclasses should implement handleAction for step-specific logic.
+   */
+  @Override
+  public JSONObject doAction( String method, TransMeta transMeta, Map<String, String> queryParams ) {
```

Comment:
would it be worth it to create unit test for the abstract class BaseStepHelper?

I see that you do have TransExecutionHelper that extends BaseHelper

- https://www.baeldung.com/junit-test-abstract-class